jupyter_server/services/kernels/kernelmanager.py

```diff
@@ -300,7 +300,7 @@ def shutdown_kernel(self, kernel_id, now=False):
             type=self._kernels[kernel_id].kernel_name
         ).dec()
 
-        return run_sync(self.super.shutdown_kernel(self, kernel_id, now=now))
+        return self.super.shutdown_kernel(self, kernel_id, now=now)
```

Comment:
Let's update the signature to include `restart=False` and convey that parameter to the superclass.